packages/typespec-client-generator-core/src/public-utils.ts

```diff
@@ -604,3 +605,40 @@ export function getHttpOperationWithCache(
   context.httpOperationCache.set(operation, httpOperation);
   return httpOperation;
 }
+
+/**
+ * Determines if a type is nullable.
+ * @param type
+ * @returns
+ */
+export function isNullable(type: SdkType): boolean {
+  if (type.kind !== "union") return false;
+  let nullCount = 0;
+  let nonNullCount = 0;
+  for (const value of type.values) {
+    if (value.kind === "null") {
+      nullCount++;
+    } else {
+      nonNullCount++;
+    }
+  }
+  return nullCount > 0 && nonNullCount > 0;
+}
+
+/**
+ * Since we don't remove null types from the values of a union type, this helper function helps return the type without null unioned in.
+ *
+ * Only call after first making sure the type isNullable with our helper function
+ * @param type
+ */
+export function getUnderlyingNullableType(type: SdkType): SdkType {
```

Comment:
Should we call this method `getNonNullType()`? `getUnderlyingType` seems like we are returning a base type or some internal type.

Reply:
this method goes away with @m-nash's proposal, since the `SdkNullableType.valueType` will get you the same value